test/simulation.jl

```diff
@@ -0,0 +1,200 @@
+    @test sim_done.elapsed_time ≥ 0.0
+end
+
+@testset "FixedFractionalCatchup" begin
+    Random.seed!(12345)
+
+    L = 5
+    statedefn = Ising2D(L; periodic=false)
+    prob = WangLandauProblem(statedefn)
+
+    ffc = WangLandau.FixedFractionalCatchup(0.25)
+    sim = CommonSolve.init(prob; catchup_strategy = ffc)
+    sim.logdos .= [1e-8, 2e-8, 3e-8, 4e-8][mod1.(1:length(sim.logdos), 4)]
+
+    oldmin = ffc.minval
+    WangLandau.update!(ffc, sim)
+    @test ffc.minval != oldmin
+    @test WangLandau.catchup_enabled(ffc) == true
+    @test WangLandau.catchup_value(ffc) ≈ ffc.minval * ffc.fraction
+
+    state, old_index = initialise_state(statedefn)
+    histogram = zeros(Int, WangLandau.histogram_size(statedefn))
+    logdos = zeros(Float64, WangLandau.histogram_size(statedefn))
+    logf = 0.1
+    new_index = WangLandau.wl_trial!(state, old_index, statedefn, logdos, histogram, logf, ffc)
+    @test 1 ≤ new_index ≤ length(histogram)
+end
+
+@testset "DynamicFractionalCatchup" begin
+    Random.seed!(1234)
+
+    L = 5
+    statedefn = Ising2D(L; periodic = false)
+    prob = WangLandauProblem(statedefn)
+
+    dfc = WangLandau.DynamicFractionalCatchup()
+    sim = CommonSolve.init(prob; catchup_strategy = dfc)
+    sim.logf_strategy = WangLandau.ReduceByFactor(; final = 1e-6)
+    sim.logdos .= 1e-8 .* (1:length(sim.logdos))
+
+    old_value = dfc.value
+    WangLandau.update!(dfc, sim)
+    val = WangLandau.catchup_value(dfc)
+    @test val isa Float64
+    @test val ≥ 0.0
+    @test WangLandau.catchup_enabled(dfc) == true
+
+    logdos = zeros(Float64, WangLandau.histogram_size(statedefn))
+    histogram = zeros(Int, WangLandau.histogram_size(statedefn))
+    
+    state, old_index = initialise_state(statedefn)
+    logf = 0.01
+    new_index = WangLandau.wl_trial!(state, old_index, statedefn, logdos, histogram, logf, dfc)
+    @test 1 ≤ new_index ≤ length(histogram)
+    @test any(histogram .> 0)
+    @test any(logdos .≥ 0)
+end
+
+@testset "FlatHistogramStrategy" begin
+    Random.seed!(1234)
+
+    f1 = WangLandau.FractionOfMean(0.8)
+    @test f1 isa FractionOfMean
+    @test f1.tol ≈ 0.8
+
+    @test_throws ArgumentError FractionOfMean(1.2)
+    @test_throws ArgumentError FractionOfMean(-0.5)
+    @test_throws ArgumentError FractionOfMean(0.5, -1)
+
+    f2 = WangLandau.FractionOfMean(0.8)
+    hist = zeros(Int, 5)
+    @test_throws ErrorException WangLandau.isflat(f2, hist)
+
+    hist .= [0, 0, 3, 0, 0]
+    @test WangLandau.isflat(f2, hist) == false
+
+    hist .= [5, 5, 5, 5, 5]
+    @test WangLandau.isflat(f2, hist) == true
+
+    hist .= [10, 9, 7, 6, 1]
+    @test WangLandau.isflat(f2, hist) == false
+
+    flat_strategy = WangLandau.FractionOfMean(0.8)
+    sim_dummy = (; flat_iterations=1, total_steps=10)
+    WangLandau.update!(flat_strategy, sim_dummy) 
+
+    s1 = StableNumVisits(2, 5)
+    @test s1 isa StableNumVisits
+    @test s1.min == 2
+    @test s1.checksteps == 5
+
+    s1.numvisits = 1
+    s1.stablesteps = 10
```

Comment:
```suggestion
    s1 = StableNumVisits(1, 10)
```
use the constructor instead of altering fields directly

Reply:
Fixed